tests/e2e-test/internal/runner/runner.go

```diff
@@ -0,0 +1,180 @@
+package runner
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+// Runner runs an e2e test
+type Runner struct {
+	pkgPath string
+	pkgName string
+	network bool
+}
+
+const (
+	expectedDir          string = ".expected"
+	expectedExitCodeFile string = "exitcode.txt"
+	expectedResultsFile  string = "results.yaml"
+	expectedDiffFile     string = "diff.patch"
+)
+
+// NewRunner returns a new runner for pkg
+func NewRunner(pkg string, network bool) (*Runner, error) {
+	info, err := os.Stat(pkg)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open path %s: %w", pkg, err)
+	}
+	if !info.IsDir() {
+		return nil, fmt.Errorf("path %s is not a directory", pkg)
+	}
+	return &Runner{
+		pkgPath: pkg,
+		pkgName: filepath.Base(pkg),
+		network: network,
+	}, nil
+}
+
+// Run runs the test
```

Comment:
Can this run multiple e2e tests concurrently?

Reply:
I think it can. Will try it.